packages/cadl-ranch-specs/http/encode/numeric/main.tsp

```diff
@@ -62,4 +62,29 @@ namespace Property {
     @encode(string)
     value?: uint32;
   }
+
+  @route("/uint8")
+  @scenario
+  @scenarioDoc("""
+    Test operation with request and response model contains property of uint8 type with string encode.
+    Expected request body:
+    ```json
+    {
+      "value": "255"
+    }
+    ```
+    Expected response body:
+    ```json
+    {
+      "value": "255"
+    }
+    ```
+    """)
+  @post
+  op uint8AsStringOptional(@body body: Uint8AsStringProperty): Uint8AsStringProperty;
```

Comment:
do you want to add one for `uint8AsString`? Not sure if it's needed, it's just that corresponds more directly with what the tests are for `uint32`. They can also be templatized this way, but you can do that in a separate pr

Reply:
the reason I add `uint8` is because .net generator doesn't support uint32 yet, so I need an `uint8` case

Reply:
I think Issabella is only about the "optional" part.

I've added uint32 as optional property, because Java type is different on required/optional for primitive type. You may not need this "optional" for uint8

Reply:
OK. I've templatize the test cases.